pkg/providers/imagefamily/bootstrap/aksbootstrap.go

```diff
@@ -420,6 +420,28 @@ func kubeBinaryURL(kubernetesVersion, cpuArch string) string {
 	return fmt.Sprintf("%s/kubernetes/v%s/binaries/kubernetes-node-linux-%s.tar.gz", globalAKSMirror, kubernetesVersion, cpuArch)
 }
 
+// CredentialProviderURL returns the URL for OOT credential provider,
+// or an empty string if OOT provider is not to be used
+func CredentialProviderURL(kubernetesVersion, arch string) string {
+	minorVersion := semver.MustParse(kubernetesVersion).Minor
+	if minorVersion < 30 { // use from 1.30; 1.29 supports it too, but we have not fully tested it with Karpenter
+		return ""
+	}
+
+	// credential provider has its own release outside of k8s version, and there'll be one credential provider binary for each k8s release,
+	// as credential provider release goes with cloud-provider-azure, not every credential provider release will be picked up unless
+	// there are CVE or bug fixes.
```

Comment:
To clarify this comment. Each k8s version will have a binary. However, we are just linking to the minor version's binary, unless there are CVE, or bug fixes? Or will the released binary get replaced still getting us the CVE/bug fixes?

Reply:
This comment is copied from elsewhere, but my expectation is that, on CVEs and bug fixes, the next patch version of binary (for each affected minor version) will be released, and we will need to bump the patch version(s) here to pick it up.